### ISSUE TITLE
Support variable length in `Generate Random String`

### ISSUE THREAD

**Issue by asimell** [robotframework/robotframework#4133]
`Generate Random String` only supports a static length for a random string. It would be nice if if you could give a range of values how long the random string should be.

E.g. something like this
```
# Assume the following variable has been set: @{values}    5    10
${str}=    Generate Random String    length=${values}
${str}=    Generate Random String    length=[5, 10]
```

**Comment by pekkaklarck** [comment]
Sounds good to me in general. Not sure is an actual list or list-literal syntax best way to give ranges, though. Couldn't it be just `5-10` or `5..10`?

**Comment by asimell** [comment]
I thought about this as well. `5-10` would be the most intuitive IMO, but it could possibly support a list as well, instead of a list-like string.

So

```
Generate Random String    length=5
Generate Random String    length=${values}
Generate Random String    length=5-10
```

I have a feeling supporting a list and a string in format `<int>-<int>` is fairly straightforward to implement.

**Comment by pekkaklarck** [comment]
Supporting actual lists is easy but I'm not sure is it worth the extra work needed for testing and documentation. Is there some use case where it would be better than simple `5-10`?

**Comment by asimell** [comment]
No direct use cases come to mind. It can probably be ignored for now and added later on if there is bigger need?

**Comment by pekkaklarck** [comment]
Yeah, adding later if needed is always easy.

**Comment by pekkaklarck** [comment]
This was implemented in PR #4138 by @asimell. Thanks for a contribution!
